discretisedfield/mesh.py

```diff
@@ -950,6 +950,179 @@ def plane(self, *args, n=None, **kwargs):
+            points to be selected from the mesh.
+
+        Returns
+        -------
+        discretisedfield.Mesh
+
+            An extracted mesh.
+
+        Examples
+        --------
+        1. Extracting the mesh at a specific point (``y=1``).
+
+        >>> import discretisedfield as df
+        ...
+        >>> p1 = (0, 0, 0)
+        >>> p2 = (5, 5, 5)
+        >>> cell = (1, 1, 1)
+        >>> mesh = df.Mesh(p1=p1, p2=p2, cell=cell)
+        ...
+        >>> plane_mesh = mesh.sel(y=1)
+
+        2. Extracting the xy-plane mesh at the mesh region center.
+
+        >>> plane_mesh = mesh.sel('z')
+
+        3. Specifying a range of points along axis ``x`` to be selected from mesh.
+
+        >>> plane_mesh = mesh.sel(x=(2, 4))
+
+        """
+        if len(args) > 1 or len(kwargs) > 1:
+            raise ValueError("Select method only accepts one dimension at a time.")
+
+        if args and not kwargs:
+            dim = args[0]
+            range_ = None
+        elif not args and kwargs:
+            dim, range_ = list(kwargs.items())[0]
+        else:
+            raise ValueError(
+                "Either one positional argument or a keyword argument can be passed."
+            )
+
+        dim_index = self.region._dim2index(dim)
+
+        if range_ is not None:
+            if isinstance(range_, numbers.Real):
+                # TODO: Some book-keeping in future.
+                selected_value = range_
+            elif isinstance(range_, (tuple, list, np.ndarray)):
+                if len(range_) != 2:
+                    raise ValueError(
+                        "The points along the selected dimension must have two"
+                        " real numbers."
+                    )
+                elif not all(isinstance(point, numbers.Real) for point in range_):
+                    raise TypeError(
+                        f"The elements of {type(range_)} passed as the value of keyword"
+                        " argument must be real numbers."
+                    )
+            else:
+                raise TypeError(
+                    "The value passed to selected dimension must be a tuple, list,"
+                    " array or real number."
+                )
+        else:
+            selected_value = self.region.center[dim_index]
+
+        sub_region = dict()
+        if range_ is None or isinstance(range_, numbers.Real):
+            idxs = [i for i in range(self.region.ndim) if i != dim_index]
+            p_1 = list()
+            p_2 = list()
+            cell = list()
+            dims = list()
+            units = list()
+            for j in idxs:
+                p_1.append(self.region.pmin[j])
+                p_2.append(self.region.pmax[j])
+                cell.append(self.cell[j])
+                dims.append(self.region.dims[j])
+                units.append(self.region.units[j])
+            tol = self.region.tolerance_factor
```

Comment:
I would move this to a separate `region.sel` method (could maybe also be private). Then we can re-use the same method here and for the subregions.

Reply:
As we do not have a cell in the `Region` only `args` would make sense (and I would probably use a private method).

Reply:
Sounds good. I can create a new PR based on `sel_mesh` branch and add it there or, do you prefer to do this from the `master`?